### ISSUE TITLE
Программа приводит к зависанию MS Word / Outlook

### ISSUE THREAD

**Issue by ascold-pl** [BladeMight/Mahou#161]
Переключение раскладки (назначена клавиша Break) регулярно (примерно 1 раз на 10 переключений) в программе MS Word приводит к зависанию последней, помогает только снятие процесса. С MS Outlook ситуация похожая, только здесь программа сама "отвисает" через несколько минут.

Windows 10 Pro (x64), версия 1709, сборка ОС 16299.192
Пакет MS Office 365 16.0.8827.2148
MS Word / MS Outlook: версии 1712 (сборка 8827.2148 Click and Run)
NET Framework 2.0, 3.0, 3.5, 4.0, 4.7.1
Mahou 2.5.6.7

Программка очень нравится, но из-за этого бага работать невозможно...

Спасибо!

**Comment by BladeMight** [comment]
@ascold-pl По поводу Word и похожих уже писали в #133 #73 #74 теперь есть [страничка на wiki](https://github.com/BladeMight/Mahou/wiki/%D0%9F%D1%80%D0%BE%D0%B1%D0%BB%D0%B5%D0%BC%D1%8B-%D0%B2-%D0%BD%D0%B5%D0%BA%D0%BE%D1%82%D0%BE%D1%80%D1%8B%D1%85-%D0%BF%D1%80%D0%BE%D0%B3%D1%80%D0%B0%D0%BC%D0%BC%D0%B0%D1%85%3F).

**Comment by ascold-pl** [comment]
Спасибо большое за инфу, попробую эмуляцию, о результатах отпишусь на следующей неделе!

P.S. Замечу ли я как простой пользователь разницу между традиционным переключением раскладки и эмуляцией? И данная эмуляция при включении опции действует для всех программ, или только для "проблемных"?

**Comment by BladeMight** [comment]
> Замечу ли я как простой пользователь разницу 

Думаю что нет.

> для всех программ

Да для всех.

> только для "проблемных"?

Можно сделать отдельную функцию типа "Использовать эмуляцию в проблемных приложениях".

**Comment by ascold-pl** [comment]
Спасибо еще раз, теперь за оперативные ответы! Вынесу на всякий случай отдельным комментарием последний вопрос:

Подскажите еще, пожалуйста, правильно ли я понимаю, что в поле "Тип эмуляции" лучше задать сочетание клавиш отличное от стандартной (скажем, оставить Alt+Shift "системным" переключением и задать Ctrl+Shift для эмуляции)?

**Comment by BladeMight** [comment]
> лучше задать сочетание клавиш отличное от стандартной 

НЕТ, тип эмуляции должен быть **такой же** как в системе, т.к. эмуляция это _нажатие этих клавиш для переключения раскладки._ Если же убрать системный Alt+Shift то переключение раскладки эмуляцией перестанет работать.

**Comment by rg-software** [comment]
Ещё по поводу эмуляции. У меня в системе стоит японская раскладка, которая активируется чуть медленнее прочих (т.к. открываются ещё дополнительные значки IME). И есть горячая клавиша, которая с помощью эмуляции переключает "рус/англ". Получается, что одно из этих нажатий должно приводить к двойному срабатыванию: рус -> яп -> англ. 

Так вот, частенько Mahou эмулирует нажатия слишком быстро для моей системы (как я это понимаю), и активной остаётся японская раскладка. Можно как-то настроить задержку между нажатиями, чтобы такое не происходило?

**Comment by BladeMight** [comment]
> которая активируется чуть медленнее прочих

Интересно... Как это? Только визуально? Или и реально?

> т.к. открываются ещё дополнительные значки IME

Если Вы про значки которые визуально появляются позже, то причина не в них, т.к. функции IME загружаются сразу же после смены раскладки, а значки могут появляться чуть позже, но это только визуально.

> И есть горячая клавиша, которая с помощью эмуляции переключает "рус/англ".

Не понял, что одна гор. клавиша для смены в двух раскладках? Если да, то такой функции в Mahou нет... Если нет, объясните что имели в виду.

> Так вот, частенько Mahou эмулирует нажатия слишком быстро для моей системы

Скорее всего проблема в порядке раскладок, прочитай https://github.com/BladeMight/Mahou/issues/180#issuecomment-405162813

P.S. У меня тоже стоит японская раскладка и используя ее не вижу смысла держать англ. раскладку все еще в списке раскладок, т.к. в Google Japanese IME, Baidu Japanese IME или Microsoft Japanese IME есть удобные гор. клавиши для переключения методов ввода, т.е. режим катаканы, хираганы и прямой ввод(Direct Input, т.е. вместь англ. раскладки). А чаще всего я между ними переключаюсь используя <kbd>Alt</kbd>+<kbd>`</kbd>(Тильда, Ё), переключая между вводом хираганы и прямым вводом.
У тебя какой IME?

**Comment by rg-software** [comment]
> Интересно... Как это? Только визуально

Это едва заметный эффект, но в OneNote, например, Mahou почти всегда мне даёт неправильную раскладку.

> Если нет, объясните что имели в виду.

У меня четыре раскладки: ru/en/fi/ja. Почти всегда бывают нужны ru/en, остальные две редко. Соответственно, я использую вот такие настройки:
![clipboard01](https://user-images.githubusercontent.com/13718460/44007175-f19c2e58-9ecb-11e8-8ae6-d3c2de0eec77.png)
Caps lock переключает туда-сюда между основными языками, а другие открываю с помощью обычных Windows клавиш (Alt+Shift). Почти всегда это работает хорошо. Но вот в OneNote почему-то по Caps lock открывается непредсказуемая раскладка.

> У тебя какой IME?

Стандартный Windows. Бывает нужно редко, поэтому я пока мало углублялся в тему различий... иногда бывает надо набрать пару слов, вот и держу.
